File: test/core/format/safe_filename_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:microvise_crm/core/format/safe_filename.dart';

void main() {
  group('safeDownloadFilename Turkish fold', () {
    test('GÜZBEL keeps U instead of underscore', () {
      expect(
        safeDownloadFilename(
          'GÜZBEL YATIRIM LTD._2026-1-00000000085.pdf',
        ),
        'GUZBEL_YATIRIM_LTD._2026-1-00000000085.pdf',
      );
    });

    test('folds mixed Turkish letters and preserves case', () {
      expect(
        foldTurkishAsciiPreserveCase('Çiğköfte ŞÖLEN'),
        'Cigkofte SOLEN',
      );
    });
  });
}

File: test/safe_filename_test.js
const { test } = require('node:test');
const assert = require('node:assert/strict');

const {
  foldTurkishAsciiPreserveCase,
  safeDownloadFilename,
  safeFilenamePart,
} = require('../api/_lib/safe_filename');

test('GÜZBEL PDF adı Ü harfini U yapar, alt çizgiye çevirmez', () => {
  assert.equal(
    safeDownloadFilename('GÜZBEL YATIRIM LTD._2026-1-00000000085.pdf'),
    'GUZBEL_YATIRIM_LTD._2026-1-00000000085.pdf',
  );
  assert.equal(safeFilenamePart('GÜZBEL'), 'GUZBEL');
  assert.equal(foldTurkishAsciiPreserveCase('GÜZBEL'), 'GUZBEL');
});
